### ISSUE TITLE
Navigation should draw a better distinction between default/intrinsic and custom labels

### ISSUE THREAD

**Issue by zerocrates** [omeka/omeka-s#631]
Right now, anything that has a default label is just directly setting that as "label" for jsTree. This means that when you create, for example, a page link, the label gets "locked in" to whatever the page's title was at the time you added the link. Changing the page title thereafter requires actually editing the page, then going back and either deleting/re-adding the link or editing the label there as well.

Instead we should draw a distinction between the user-entered label (appearing in the detailed form for a nav link) and the actual label. A link may provide some default in case the user doesn't specify a label, but that shouldn't result in that default being treated like a user-set custom label.

In other words, only labels explicitly set by the user should be "sticky." Ones that are being provided by default should be dynamic, so they can adjust with things like changing locales (that would change the translation of the default label) or changing data (like the title of a linked page)

**Comment by jimsafley** [comment]
* Custom URL
  * Label
    * required
    * do not autofill
  * URL
    * required
    * do not autofill
* Browse
  * Label
    * optional
    * do not autofill
    * default: "Browse"
  * Query
    * required
    * do not autofill
* Page
  * Label
    * optional
    * do not autofill
    * default: page title
